src/main/java/frc/robot/subsystems/FourBarArm.java

```diff
@@ -14,32 +14,57 @@ public class FourBarArm extends SubsystemBase {
   private final CANSparkMax m_motor;
   private final PIDController m_pid;
   private final RelativeEncoder m_encoder;
-  private double m_armSetpoint;
 
   public FourBarArm() {
+    // configure the motor
     m_motor = new CANSparkMax(ArmConstants.motorID, MotorType.kBrushless);
     m_motor.setIdleMode(IdleMode.kBrake);
 
+    // configure the encoder
+    // TODO: use a kConstant instead of the 8192
     m_encoder = m_motor.getAlternateEncoder(SparkMaxAlternateEncoder.Type.kQuadrature, 8192);
+    // The RelativeEncoder reports angles in native revolutions by default.
+    // See https://codedocs.revrobotics.com/java/com/revrobotics/relativeencoder
+    // Change the encoder's reported value to radians (1 revolution = 2 pi radians).
     m_encoder.setPositionConversionFactor(2*Math.PI);
-    m_encoder.setVelocityConversionFactor(2*Math.PI/60);  
+    // The RelativeEncoder reports RPM by default.
+    // Change the velocity to radians per second (1 RPM = 2 pi radians / 60 seconds)
+    m_encoder.setVelocityConversionFactor(2.0 * Math.PI / 60.0);  
 
+    // make the PID controller
     m_pid = new PIDController(ArmConstants.kP, ArmConstants.kI, ArmConstants.kD);
-    m_pid.setSetpoint(ArmConstants.initialPosition);
+    // set the PID controller's tolerance
     m_pid.setTolerance(ArmConstants.kTolerance);
+    // go to the initial position (use the class method)
+    setArmSetpoint(ArmConstants.initialPosition);
   }
 
+  /**
+   * Set the FourBarArm's desired position.
+   * @param setpoint the desired arm position (in radians)
+   */
   public void setArmSetpoint(double setpoint) {
+    // set the PID integration error to zero.
     m_pid.reset();
+    // set the PID desired position
     m_pid.setSetpoint(setpoint);
-    m_armSetpoint = setpoint;
   }
 
   @Override
   public void periodic() {
-    m_motor.set(MathUtil.clamp(m_pid.calculate(m_encoder.getPosition()), ArmConstants.minMotorPower, ArmConstants.maxMotorPower));
+    // calculate the PID power level
+    double pidPower = m_pid.calculate(m_encoder.getPosition());
+    // calculate the feedforward power (nothing for now)
+    double feedforwardPower = 0.0;
+
+    // set the motor power
+    m_motor.set(MathUtil.clamp(pidPower + feedforwardPower, ArmConstants.minMotorPower, ArmConstants.maxMotorPower));
```

Comment:
I don't think this is necessary. Yichen is working on the four bar arm feedforward. Please talk to us if you want to suggest changes like this.

Reply:
Bare code is often difficult to understand. I've added comments that explain the research I did to understand the code.

For PIDF, WPILIB has bottled up the PID but not the F. Separate them to make the pieces clear.

Reply:
Yes, I don't think the comments are really necessary to explain that research. We have the WPI documentation, and I did some work to explain PID + FF to the students during last semester.